lib/cocoapods/generator/copy_xcframework_script.rb

```diff
@@ -93,11 +92,31 @@ def script
     target_variant="maccatalyst"
   fi
   for i in ${!paths[@]}; do
-    if [[ "${paths[$i]}" == *"$target_arch"* ]] && [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+    local matched_all_archs="1"
+    for target_arch in $target_archs
+    do
+      if ! [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+        matched_all_archs="0"
+        break
+      fi
+
+      # This regex matches all possible variants of the arch in the folder name:
+      # Let's say the folder name is: ios-armv7_armv7s_arm64_arm64e/CoconutLib.framework
+      # We match the following: -armv7_, _armv7s_, _arm64_ and _arm64e/.
+      # If we have a specific variant: ios-i386_x86_64-simulator/CoconutLib.framework
+      # We match the following: -i386_ and _x86_64-
+      local target_arch_regex="[_\\-]${target_arch}[\\/_\\-]"
```

Comment:
I think we might need to account for something like `-ios-armv7/CoconutLib.framework`, so should the second `[_\-]` get `/` added as well?

Reply:
I'm not sure what you are talking about? That regex also matches `-armv7/`

Reply:
ah you're right, sorry the escapes through me off 👍